sys/include/net/netopt.h

```diff
@@ -269,6 +269,18 @@ typedef enum {
      */
     NETOPT_LAST_ED_LEVEL,
 
+    /**
+     * @brief   Type for sequiential processing of IPv6 extension headers
+     *
+     * On GET it requests a thread to provide an extension header.
+     * On SET it always returns -ENOTSUP.
+     *
+     * When the header is done the thread replies to the call.
+     * The result is the pktsnip_t with the extension header
+     * The result is NULL if no extension header is passed.
```

Comment:
> not a pointer of a pointer

true, it would require some wrapping.
You mean exploiting the int return type for pointer passing, why not.